systemtest/gencorpora/catbulk.go

```diff
@@ -113,18 +171,40 @@ func handleReq(writer io.Writer) http.HandlerFunc {
 
 			mockResp := esutil.BulkIndexerResponse{}
 			scanner := bufio.NewScanner(reader)
+
+			var stat docsStat
+			var buf bytes.Buffer
 			for count := 0; scanner.Scan(); count++ {
-				fmt.Fprintln(writer, scanner.Text())
+				if _, err := fmt.Fprintln(&buf, scanner.Text()); err != nil {
```

Comment:
```suggestion
				if _, err := buf.Write(scanner.Bytes()); err != nil {
```

**suggestion**: I think we can avoid a byte->string conversion and use the buffer directly. WDYT ?

Reply:
This will not work since the bulk API requires newline delimited JSON and scanner strips EOL markers. However, it is possible to optimize performance here by customizing the scanner -- I was being lazy here since it is not in the critical code path. Let me update this.